schema/build-config-schema.js

```diff
@@ -343,15 +361,17 @@ rules.forEach(function forRule(rule) {
           "description": "List of headings",
           "type": "array",
           "items": {
-            "type": "string"
+            "type": "string",
+            "pattern": "^(\\*|\\+|#{1,6} .*)$"
```

Comment:
Good idea! For ` .*` (includes space), could it be `\\s\\S.*` instead? In other words, single space then non-space then anything.

Reply:
I am not sure whether it's a good idea according to [this note](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html):

> The regular expression syntax used is from JavaScript (ECMA 262, specifically). However, that complete syntax is not widely supported, therefore it is recommended that you stick to the subset of that syntax described below.